### PR TITLE
B 20529 INT add save retrieve and filter capability for assigned column sc

### PR DESCRIPTION
**Pull request by joeydoyecaci**
## Summary
Adds in Assigned User Column on Service Counselors Queue with the ability to filter, sort, and assign moves to a Service Counselor.

### How to test as SC
1. Log in as a Service Counselor
2. Click on the dropdown of any move under Assigned, and select a user from the dropdown
3. Repeat step 2 for another user
4. Refresh the page
5. Enter the name of the user that was assigned for either of the moves in the Assigned User column header
6. There should be a pill off to the left of the top of the table with the name of the user filtered and an 'X', only moves that are assigned to the user entered into Assigned User as a filter should display.
7. Click the 'X' button for the filter to remove this filter. After doing so, all moves should show regardless of assigned
8. Click on the Assigned Column to sort in Ascending order, moves should show based on the name of who's assigned
9. Click again to show assigned users in Descending order, moves should show like before but in descending order
10. Click once more to show assigned moves in unordered mode.

### CODE REVIEW

File: pkg/services/move.go [hunk 1]
```diff
@@ -120,7 +120,6 @@ type MoveCloseoutOfficeUpdater interface {
 type MoveCanceler interface {
 	CancelMove(appCtx appcontext.AppContext, moveID uuid.UUID) (*models.Move, error)
 }
-
```
Comment (JonSpight) [inline pkg/services/move.go:123]: I would revert this change, all the other types have separator between them.

File: src/components/Table/TableQueue.jsx [hunk 2]
```diff
@@ -119,6 +123,14 @@ const TableQueue = ({
     [],
   );
   const tableData = useMemo(() => data, [data]);
+  const formattedAvailableOfficeUsers = formatAvailableOfficeUsers(availableOfficeUsers, isSupervisor, currentUserId);
+  // attach the available office users to the moves/row
+  const tableDataWithAvailableUsers = tableData?.map((ele) => {
+    const newEle = { ...ele };
```
Comment (JonSpight) [inline src/components/Table/TableQueue.jsx:129]: ele and newEle seems a bit ambiguous to me, can you make them more detailed? maybe row and newRow?

**Review by JonSpight — COMMENTED** [review]
Filtering on the assigned column works if you type in the entire First name or Last name.
for example "Ryan" will not be found if "ry" Is searched. 

Does anyone have thoughts on making the assigned column include partial searches. (ex. Customer name and Counseling Office columns)

File: src/components/Table/TableQueue.jsx (hunk 2 above)
Comment (paulstonebraker) [reply src/components/Table/TableQueue.jsx:129]: I think this was from my initial commits before I passed to Joey? But I agree

File: pkg/services/move.go (hunk 1 above)
Comment (joeydoyecaci) [reply pkg/services/move.go:123]: Done

**Comment by joeydoyecaci** [comment]
> Filtering on the assigned column works if you type in the entire First name or Last name. for example "Ryan" will not be found if "ry" Is searched.
> 
> Does anyone have thoughts on making the assigned column include partial searches. (ex. Customer name and Counseling Office columns

To do so would conflict with the delimiting function in place to allow searching between first and last name with the comma in place. While it would be optimal to add this in place, it would not be feasible.

File: src/components/Table/TableQueue.jsx (hunk 2 above)
Comment (joeydoyecaci) [reply src/components/Table/TableQueue.jsx]: they were not changed by me, but renamed nonetheless.

**Review by paulstonebraker — APPROVED** [review]
Code looks good, works as intended.

Some changes outside the scope of this work to move_factory.go, move_test.go, mto_service_items_test.go, and ghc_api.js; nothing crazy and probably things brought over during a merge. I won't block approval of this for those, but consider reverting then.

**Comment by joeydoyecaci** [comment]
@paulstonebraker I'll still see about removing those. Those changes were merge related. Just wasn't 100% sure on if those were outside updates from main that should go or not.

**Comment by joeydoyecaci** [comment]
@paulstonebraker quick follow up: 
ghc_api.js : merging keeps shifting those functions around. I've just moved them to the bottom just to help stop it from happening.
mto_service_items_test.go: those were go linter related issues. there were multiple references to the same go package. was trying to give the pre-commit hooks one less reason to block the commit by consolidating the multiple references to one reference to one package
move_test.go: same as above
move_factory.go: merge related whitespace removal. will remove in next commit with any further changes needed.

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload.go [hunk 3]
```diff
@@ -2034,7 +2034,7 @@ func QueueAvailableOfficeUsers(officeUsers []models.OfficeUser) *ghcmessages.Ava
 }
 
 // QueueMoves payload
-func QueueMoves(moves []models.Move) *ghcmessages.QueueMoves {
+func QueueMoves(moves []models.Move, officeUsers []models.OfficeUser, officeID uuid.UUID) *ghcmessages.QueueMoves {
```
Comment (pambecker) [inline pkg/handlers/ghcapi/internal/payloads/model_to_payload.go:2037]: Is officeID needed here?  doesn't seem to ever be used.

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload.go [hunk 4]
```diff
@@ -2184,7 +2185,7 @@ func queuePaymentRequestStatus(paymentRequest models.PaymentRequest) string {
 }
 
 // QueuePaymentRequests payload
-func QueuePaymentRequests(paymentRequests *models.PaymentRequests) *ghcmessages.QueuePaymentRequests {
+func QueuePaymentRequests(paymentRequests *models.PaymentRequests, officeUsers []models.OfficeUser, officeID uuid.UUID) *ghcmessages.QueuePaymentRequests {
```
Comment (pambecker) [inline pkg/handlers/ghcapi/internal/payloads/model_to_payload.go:2188]: same here, Is officeID needed here?  doesn't seem to ever be used.

Comment (paulstonebraker) [reply pkg/handlers/ghcapi/internal/payloads/model_to_payload.go:2188]: that was my work... let me think about it 🤔

Comment (paulstonebraker) [reply pkg/handlers/ghcapi/internal/payloads/model_to_payload.go]: @pambecker updated

**Review by pambecker — APPROVED** [review]

**Review by JonSpight — APPROVED** [review]
